tests/forking_agent/test_agent_executor.py

```diff
@@ -312,3 +315,84 @@ def test():
+
+
+async def test_executor_creation_and_venv_usage(pip_index, mpmanager_light, monkeypatch) -> None:
+    """
+    This test verifies the creation and reuse of executors based on their blueprints. It checks whether
+    the concurrency aspects and the locking mechanisms work as intended.
+    """
+
+    requirements1 = ()
+    requirements2 = ("pkg1",)
+    pip_config = PipConfig(index_url=pip_index.url)
+
+    # Prepare a source module and its hash
+    code = """
+def test():
+    return 10
+    """.encode()
+    sha1sum = hashlib.new("sha1")
+    sha1sum.update(code)
+    hv: str = sha1sum.hexdigest()
+    module_source1 = ModuleSource(
+        name="inmanta_plugins.test",
+        hash_value=hv,
+        is_byte_code=False,
+        source=code,
+    )
+    sources1 = ()
+    sources2 = (module_source1,)
+
+    blueprint1 = executor.ExecutorBlueprint(pip_config=pip_config, requirements=requirements1, sources=sources1)
+    blueprint2 = executor.ExecutorBlueprint(pip_config=pip_config, requirements=requirements2, sources=sources2)
+
+    executor_manager = mpmanager_light
+    executor_1, executor_2 = await asyncio.gather(
+        executor_manager.get_executor("agent1", "local:", code_for(blueprint1), 0.1),
+        executor_manager.get_executor("agent2", "local:", code_for(blueprint2), 0.1),
+    )
+
+    old_datetime = datetime.datetime(year=2022, month=9, day=22, hour=12, minute=51, second=42)
+    os.utime(
+        f"{executor_2.executor_virtual_env.env_path}/{const.INMANTA_ENV_STATUS_FILENAME}",
+        (old_datetime.timestamp(), old_datetime.timestamp()),
+    )
+
+    def get_modification_datetime(file: str) -> datetime.datetime:
+        return datetime.datetime.fromtimestamp(os.stat(file).st_mtime)
+
+    old_check_executor1 = get_modification_datetime(
+        f"{executor_1.executor_virtual_env.env_path}/{const.INMANTA_ENV_STATUS_FILENAME}"
+    )
+    old_check_executor2 = get_modification_datetime(
+        f"{executor_2.executor_virtual_env.env_path}/{const.INMANTA_ENV_STATUS_FILENAME}"
+    )
+
+    await asyncio.sleep(0.2)
+
+    new_check_executor1 = get_modification_datetime(
+        f"{executor_1.executor_virtual_env.env_path}/{const.INMANTA_ENV_STATUS_FILENAME}"
+    )
+    new_check_executor2 = get_modification_datetime(
+        f"{executor_2.executor_virtual_env.env_path}/{const.INMANTA_ENV_STATUS_FILENAME}"
+    )
+
+    assert new_check_executor1 > old_check_executor1
+    assert new_check_executor2 > old_check_executor2
+    assert (datetime.datetime.now() - new_check_executor2).seconds <= 2
+
+    monkeypatch.undo()
+
+    # Now we want to check if the cleanup is working correctly
+    await executor_manager.stop_for_agent("agent1")
+    # First we want to override the modification date of the `inmanta_env_status` file
+    os.utime(
+        f"{executor_1.executor_virtual_env.env_path}/.inmanta_env_status", (old_datetime.timestamp(), old_datetime.timestamp())
```

Comment:
By updating both the modification and the access timestamp we lose test coverage. If the `clean_environments()` method only takes into account the access timestamp, this test case will succeed, but in production we will get different behavior, because the code that touches the file only updates the modification timestamp.